tools/attach_logs_in_jira.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+import sys
+import os
+import argparse
+import logging
+import tarfile
+from pathlib import Path
+from functools import partial
+
+import jira
+
+from consts import JIRA_PROJECT, JIRA_SUMMARY, JIRA_SERVER
+
+JIRA_ATTACHMENT_MAX_SIZE = 50 * 2**20  # 50MiB
+
+log = logging.getLogger(__name__)
+stream_handler = logging.StreamHandler(sys.stderr)
+stream_handler.setFormatter(logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s"))
+log.addHandler(stream_handler)
+
+
+def _get_or_create_jira_ticket(jira_client: jira.JIRA, summary: str) -> jira.Issue:
+    matching_issues = jira_client.search_issues(
+        f'project = "{JIRA_PROJECT}" AND summary ~ "{summary}"', fields=["attachment"]
+    )
+
+    if len(matching_issues) == 0:
+        log.info("Creating a placeholder issue for %s", summary)
+        return jira_client.create_issue(
+            project=JIRA_PROJECT,
+            summary=summary,
+            description=(
+                "Logs are still being uploaded.\nTicket description and signatures will get processed shortly..."
+            ),
+        )
+
+    if len(matching_issues) == 1:
+        log.debug("Using pre-existing issue %s", matching_issues[0])
+        return matching_issues[0]
+
+    raise RuntimeError(
+        f"Found more than one matching issue for '{summary}': "
+        f"{', '.join(issue.key for issue in matching_issues)}"
+    )
+
+
+def _archive_dir(directory: Path) -> Path:
+    archived_file = directory.with_suffix(".tar.gz")
+    log.info("Archiving directory %s to %s", directory, archived_file)
+    with tarfile.open(archived_file, "w:gz") as tar:
+        tar.add(directory, directory.name)
+
+    return archived_file
+
+
+def _split_file(path: Path, chunk_size: int) -> None:
+    log.info("Splitting %s to several files", path)
+    with path.open("rb") as whole_file:
+        read_chunk = partial(whole_file.read, chunk_size)
+        for i, chunk in enumerate(iter(read_chunk, b"")):
+            with open(f"{path}-part-{i}", "wb") as part_file:
+                part_file.write(chunk)
+
+
+def attach_logs_in_jira(source_dir: Path, jira_access_token: str):
+    jira_client = jira.JIRA(JIRA_SERVER, token_auth=jira_access_token, validate=True)
+
+    for cluster_logs_dir in source_dir.iterdir():
+        issue = _get_or_create_jira_ticket(jira_client, JIRA_SUMMARY.format(failure_id=cluster_logs_dir.name))
```

Comment:
```suggestion
        issue = _get_or_create_jira_ticket(jira_client, JIRA_SUMMARY.format(failure_id=cluster_logs_dir.name))
```
in order to make the process resilient, it could be good to catch the possible exception thrown by `_get_or_create_jira_ticket`, log and process the next folder

Reply:
Implemented something for it. I think in another PR I'll upgrade to python 3.11 to make use of ``ExceptionGroup`` (instead of how I do it with current code)

Reply:
https://github.com/openshift-assisted/assisted-installer-deployment/pull/344